components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.cpp

```diff
@@ -495,6 +514,41 @@ void SHOCMacrophysics::run_impl (const double dt)
                        default_policy,
                        shoc_postprocess);
   Kokkos::fence();
+
+  // Extra SHOC output diagnostics
+  if (m_params.get<bool>("extra_shoc_diags", false)) {
+
+    const auto& shoc_mix = get_field_out("shoc_mix").get_view<Spack**>();
+    Kokkos::deep_copy(shoc_mix,history_output.shoc_mix);
```

Comment:
This is indeed costly, for no reason. In principle, we would like history_output.shoc_mix to simply use the view stored inside the Field "shoc_mix". We are basically adding 10 3d fields for no real reason...

OTOH, this may add more complexity, since we need to be careful during init, depending on the value of `extra_shoc_diags`:
- false: request enough buffer mem to accommodate these vars.
- true: do NOT request buffer mem for these fields, and grab them from the computed field during initialize_impl

Reply:
The buffer needs a bit of a reworked to allow for it to be dynamically sized based on a param and not be unreadable. @bogensch If you do not want to do that I can take a look. But at the very least we can get rid of the deep copies like Luca said, in initialize_impl:
```
if(m_params.get<bool>("extra_shoc_diags")) {
  history_output.shoc_mix = get_field_out("shoc_mix").get_view<Spack**>();
} else {
  history_output.shoc_mix = m_buffer.shoc_mix;
}
```

Reply:
Not too much logic to add. When SHOCMacrophysics::requested_buffer_size_in_bytes and SHOCMacrophysics::init_buffers are called, we can ping m_params, to see if we need the extra mem for those fields or not.

Reply:
You're right, it's not too much logic. But we use a `constexpr` for number of fields in the buffer, and then we initialize the buffer views in a way that probably doesn't lend itself to using if-else nicely.

Reply:
@tcclevenger what's your plan regarding these unnecessary copies? Are you planning to fix them here or in another PR? If the latter, you can go ahead and merge, since the cosp failures are orthogonal.